off_chain_data/application-go/blockParser.go

```diff
@@ -0,0 +1,397 @@
+package main
+
+import (
+	"fmt"
+
+	"github.com/hyperledger/fabric-gateway/pkg/identity"
+	"github.com/hyperledger/fabric-protos-go-apiv2/common"
+	"github.com/hyperledger/fabric-protos-go-apiv2/ledger/rwset"
+	"github.com/hyperledger/fabric-protos-go-apiv2/ledger/rwset/kvrwset"
+	"github.com/hyperledger/fabric-protos-go-apiv2/msp"
+	"github.com/hyperledger/fabric-protos-go-apiv2/peer"
+	"google.golang.org/protobuf/proto"
+)
+
+type Block interface {
+	GetNumber() uint64
+	GetTransactions() []Transaction
+	ToProto() *common.Block
+}
+
+type ParsedBlock struct {
+	block           *common.Block
+	validationCodes []byte
+	transactions    []Transaction
+}
+
+func NewParsedBlock(block *common.Block) Block {
+	validationCodes := getTransactionValidationCodes(block)
+
+	return &ParsedBlock{block, validationCodes, nil}
+}
+
+func (pb *ParsedBlock) GetNumber() uint64 {
+	header := assertDefined(pb.block.GetHeader(), "missing block header")
+	return header.GetNumber()
+}
+
+// TODO: needs cache, getPayloads, parsePayload
+func (pb *ParsedBlock) GetTransactions() []Transaction {
```

Comment:
Go typically doesn't use `Get` for getters (although it can make sense in some cases). A more typical naming would just be `Transactions`. See [Effective Go](https://go.dev/doc/effective_go#Getters).

```suggestion
func (pb *ParsedBlock) Transactions() []Transaction {
```

Reply:
Yes yes, this was also bothering me. I remember that one well. Getters and setters aren't a good pattern anyway imho. I agree mostly with Allen Holub on this one => good read [Why getter and setter methods are evil](https://www.infoworld.com/article/2161183/why-getter-and-setter-methods-are-evil.html). But I would not go as far as not using them at all.